src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1108,8 +1091,6 @@ private void ScanBlank()
         /// <summary>Scans chars following a '\' (not counting the '\'), and returns a RegexNode for the type of atom scanned</summary>
         private RegexNode? ScanBackslash(bool scanOnly)
         {
-            Debug.Assert(_pos < _pattern.Length, "The current reading position must not be at the end of the pattern");
```

Comment:
Why was this removed?

Reply:
It will throw out of bounds just below. We generally don't assert indexes just before we use them.

Reply:
We do generally assert preconditions.  Calling this when there's pattern remaining is a precondition.